renderlib/RotateTool.cpp

```diff
@@ -0,0 +1,367 @@
+#include "RotateTool.h"
+
+struct ManipColors
+{
+  static constexpr glm::vec3 xAxis = { 1.0f, 0.0f, 0.0f };
+  static constexpr glm::vec3 yAxis = { 0.0f, 1.0f, 0.0f };
+  static constexpr glm::vec3 zAxis = { 0.0f, 0.0f, 1.0f };
+  static constexpr glm::vec3 bright = { 1.0f, 1.0f, 1.0f };
+};
+
+static const float s_orthogonalThreshold = cos(glm::radians(89.0f));
+
+static float
+getSignedAngle(const glm::vec3& v0, const glm::vec3& v1, const glm::vec3& vN)
+{
+  // get signed angle between the two vectors using (1,0,0) as plane normal
+  // atan2 ( (v0 X v1) dot (vN), v0 dot v1 )
+
+  float dp = dot(v0, v1);
+  float angle = atan2(dot(cross(v0, v1), vN), dp);
+  return angle;
+}
+
+static float
+getDraggedAngle(const glm::vec3& vN, const glm::vec3& p, const glm::vec3& l, const glm::vec3& l0, const glm::vec3& l1)
+{
+  glm::vec3 globalAxis = normalize(p - l);
+  bool axisIsOrthogonal = abs(dot(globalAxis, vN)) < s_orthogonalThreshold;
+
+  // axis is represented by vN, and a point in plane p (the center of our circle)
+  // line l-l0 is the ray of the initial mouse click
+  // line l-l1 is the ray of the current mouse position
```

Comment:
Is this convention for lines? Could these variables be named in a more descriptive way?

Reply:
It's a convention in this code.  The way l, l0, and l1 are set up is: l is the "camera origin", l0 is the first point clicked, and l1 is the second point (where the mouse was currently dragged to).  I hope the previous comments where those vars are defined should describe them.  
I could rename the vars tho...  I think it's easier from a "labeled geometric diagram" to use these short symbol names but maybe harder when looking at the code for the first time?

Reply:
Reading through the rest of the codebase, I think this is fine. You explain `l + l0 + l1` everywhere they're used.